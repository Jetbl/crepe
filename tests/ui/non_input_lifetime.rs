// In this test, the relation `Out` refers to data inside the `In` relation by
// taking a reference to its contents.
// Since that lifetime is related to input data which is moved and consumed
// within the `run(self)` function, the reference would out-live it's lifetime.

mod datalog {
    use crepe::crepe;

    crepe! {
        @input
        struct Input(u32, u32);

        @output
        struct Output<'a>(&'a u32);

        Output(&a) <- Input(a, _);
    }
}

fn main() {}
