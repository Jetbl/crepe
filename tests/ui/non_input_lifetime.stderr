error: Lifetime on output relation without any input relations having a lifetime
  --> tests/ui/non_input_lifetime.rs:14:22
   |
14 |         struct Output<'a>(&'a u32);
   |                      ^^^^
